ngraph/core/src/op/assign.cpp

```diff
@@ -60,22 +65,26 @@ void op::v3::Assign::validate_and_infer_types()
     auto variable_info = m_variable->get_info();
     NODE_VALIDATION_CHECK(this,
                           m_variable_id == variable_info.variable_id,
-                          "Variables identifiers are inconsistent.");
-    NODE_VALIDATION_CHECK(
-        this, arg_t == variable_info.data_type, "Variables types are inconsistent.");
+                          "Variables identifiers are inconsistent. Got: ",
+                          m_variable_id,
+                          " and ",
+                          variable_info.variable_id);
 
-    if (output_shape.is_static() && variable_info.data_shape.is_static())
-    {
-        NODE_VALIDATION_CHECK(this,
-                              output_shape == variable_info.data_shape,
-                              "Variables output shapes are inconsistent.");
+    NODE_VALIDATION_CHECK(this,
+                          arg_t == variable_info.data_type,
+                          "Variables types are inconsistent. Got: ",
+                          arg_t,
+                          " and ",
+                          variable_info.data_type);
 
-        set_output_type(0, arg_t, output_shape);
-    }
-    else
-    {
-        set_output_type(0, arg_t, PartialShape::dynamic());
-    }
+    NODE_VALIDATION_CHECK(this,
```

Comment:
This check is added to produce the most specialized output shape (with less number of dynamic dimensions as possible)